sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/AccessPointAdapter.java

```diff
@@ -229,21 +230,14 @@ record ExtRefAnalyzeRecord(List<SclReportItem> sclReportItems, List<TExtRef> tEx
      */
     public ExtRefAnalyzeRecord getAllCoherentExtRefForAnalyze() {
         List<SclReportItem> sclReportItems = new ArrayList<>();
-        List<TExtRef> tExtRefList = new ArrayList<>();
-        streamLDeviceAdapters().map(lDeviceAdapter -> {
-            List<TExtRef> extRefs = lDeviceAdapter.getLN0Adapter().getExtRefs().stream().filter(TExtRef::isSetSrcCBName).collect(Collectors.toCollection(ArrayList::new));
-            sclReportItems.addAll(checkExtRefWithoutServiceType(extRefs, lDeviceAdapter.getLN0Adapter().getXPath()));
-            extRefs.removeIf(tExtRef -> !tExtRef.isSetServiceType());
-            return extRefs;
-        }).flatMap(Collection::stream).forEach(tExtRef -> {
-            if (tExtRefList.isEmpty())
-                tExtRefList.add(tExtRef);
-            else {
-                if (tExtRefList.stream().noneMatch(t -> isExtRefFeedBySameControlBlock(tExtRef, t)))
-                    tExtRefList.add(tExtRef);
-            }
-        });
-        return new ExtRefAnalyzeRecord(sclReportItems, tExtRefList);
+        List<TExtRef> tExtRefList = streamLDeviceAdapters().map(lDeviceAdapter -> {
+                    List<TExtRef> extRefs = lDeviceAdapter.getLN0Adapter().getExtRefs().stream().filter(TExtRef::isSetSrcCBName).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
As a result if we filter duplicated ExtRefs is InputsAdapter.

```suggestion
                    List<TExtRef> extRefs = lDeviceAdapter.getLN0Adapter().getInputsAdapter().filterDuplicatedExtRefs().stream().filter(TExtRef::isSetSrcCBName).collect(Collectors.toCollection(ArrayList::new));
                                        sclReportItems.addAll(checkExtRefWithoutServiceType(extRefs, lDeviceAdapter.getLN0Adapter().getXPath()));
                    extRefs.removeIf(tExtRef -> !tExtRef.isSetServiceType());
                    return extRefs;
                }).flatMap(Collection::stream)
                .toList();
        return new ExtRefAnalyzeRecord(sclReportItems, tExtRefList);
```

Reply:
will not work beause it will not detect duplicated ExtRef between 2 LDevices

Reply:
Ok, I didn't know that was a requirement